cosmos/models/Workflow.py

```diff
@@ -400,16 +405,20 @@ def run(self, max_cores=None, dry=False, set_successful=True,
             self.log.fatal(ex, exc_info=True)
             raise
 
-    def terminate(self, due_to_failure=True):
+    def terminate(self, due_to_failure=True, is_cleanup=True):
         self.log.warning('Terminating %s!' % self)
         if self.jobmanager:
-            self.log.info('Processing finished tasks and terminating %s running tasks' % len(self.jobmanager.running_tasks))
+            self.log.info('Processing finished tasks and terminating {num_running_tasks} running tasks '
+                          'and cleaning up {num_dead_tasks} dead tasks'.format(
+                              num_running_tasks=len(self.jobmanager.running_tasks),
+                              num_dead_tasks=len(self.jobmanager.dead_tasks),
+                          ))
             _process_finished_tasks(self.jobmanager)
-            self.jobmanager.terminate()
+            self.jobmanager.terminate(is_cleanup=is_cleanup)
 
         if due_to_failure:
             self.status = WorkflowStatus.failed
-        else:
+        elif not is_cleanup:
```

Comment:
is it possible for due_to_failure=True and is_cleanup=True? if not make should there be an assertion?